From 0bd78f81d980238bb35253f2f7c547ef129dc4d8 Mon Sep 17 00:00:00 2001
From: Alex Naidis <alex.naidis@linux.com>
Date: Wed, 14 Sep 2016 20:43:19 +0200
Subject: [PATCH 7/8] build: switch ART to performance mode if possible

* This is possible if the dexpreopter is in performance mode too.

Change-Id: Id8ce78584fa6e5a300f089d7977c779e9b9057a5
Signed-off-by: Alex Naidis <alex.naidis@linux.com>

art: build: Fix defines for USE_DEX2OAT_DEBUG

* "?=" cannot be overriden by "?="

Change-Id: Ib5184d98740075e38375a2a7c27f117776717f33
Signed-off-by: Alex Naidis <alex.naidis@linux.com>
---
 build/Android.common_build.mk | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/build/Android.common_build.mk b/build/Android.common_build.mk
index f5a95fa0c..d04be8724 100644
--- a/build/Android.common_build.mk
+++ b/build/Android.common_build.mk
@@ -29,9 +29,15 @@ include art/build/Android.common.mk
 # Beware that tests may use the non-debug build for performance, notable 055-enum-performance
 #
 ART_BUILD_TARGET_NDEBUG ?= true
-ART_BUILD_TARGET_DEBUG ?= true
 ART_BUILD_HOST_NDEBUG ?= true
+
+ifneq ($(USE_DEX2OAT_DEBUG),false)
+ART_BUILD_TARGET_DEBUG ?= true
 ART_BUILD_HOST_DEBUG ?= true
+else
+ART_BUILD_TARGET_DEBUG ?= false
+ART_BUILD_HOST_DEBUG ?= false
+endif
 
 ifeq ($(ART_BUILD_TARGET_NDEBUG),false)
 $(info Disabling ART_BUILD_TARGET_NDEBUG)
-- 
2.18.0

